File: python-sdk/src/boxsdk/models/shared/skillcardsmetadata.py
import dataclasses
from typing import Any,Optional
from dataclasses_json import dataclass_json
from boxsdk import utils


@dataclass_json
@dataclasses.dataclass
class SkillCardsMetadata:
    r"""SkillCardsMetadata
    The metadata assigned to a using for Box skills.
    """
    
    dollar_can_edit: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('$canEdit') }})
    dollar_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('$id') }})
    dollar_parent: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('$parent') }})
    dollar_scope: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('$scope') }})
    dollar_template: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('$template') }})
    dollar_type: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('$type') }})
    dollar_type_version: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('$typeVersion') }})
    dollar_version: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('$version') }})
    cards: Optional[list[Any]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('cards') }})
    

File: sdks/java/lib/src/main/java/models/shared/SkillCardsMetadata.java
package .models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * SkillCardsMetadata
 * The metadata assigned to a using for Box skills.
**/
public class SkillCardsMetadata {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("$canEdit")
    public Boolean dollarCanEdit;
    public SkillCardsMetadata withDollarCanEdit(Boolean dollarCanEdit) {
        this.dollarCanEdit = dollarCanEdit;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("$id")
    public String dollarId;
    public SkillCardsMetadata withDollarId(String dollarId) {
        this.dollarId = dollarId;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("$parent")
    public String dollarParent;
    public SkillCardsMetadata withDollarParent(String dollarParent) {
        this.dollarParent = dollarParent;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("$scope")
    public String dollarScope;
    public SkillCardsMetadata withDollarScope(String dollarScope) {
        this.dollarScope = dollarScope;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("$template")
    public String dollarTemplate;
    public SkillCardsMetadata withDollarTemplate(String dollarTemplate) {
        this.dollarTemplate = dollarTemplate;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("$type")
    public String dollarType;
    public SkillCardsMetadata withDollarType(String dollarType) {
        this.dollarType = dollarType;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("$typeVersion")
    public Long dollarTypeVersion;
    public SkillCardsMetadata withDollarTypeVersion(Long dollarTypeVersion) {
        this.dollarTypeVersion = dollarTypeVersion;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("$version")
    public Long dollarVersion;
    public SkillCardsMetadata withDollarVersion(Long dollarVersion) {
        this.dollarVersion = dollarVersion;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("cards")
    public Object[] cards;
    public SkillCardsMetadata withCards(Object[] cards) {
        this.cards = cards;
        return this;
    }
}